CodeEdit/Features/Documents/WorkspaceDocument/WorkspaceDocument+Find.swift

```diff
@@ -105,24 +104,34 @@ extension WorkspaceDocument.SearchState {
         }
 
         let asyncController = SearchIndexer.AsyncManager(index: indexer)
-
         let evaluateResultGroup = DispatchGroup()
         let evaluateSearchQueue = DispatchQueue(label: "app.codeedit.CodeEdit.EvaluateSearch")
 
         let searchStream = await asyncController.search(query: searchQuery, 20)
         for try await result in searchStream {
             for file in result.results {
+                let fileURL = file.url
+                let fileScore = file.score
+                let capturedRegexPattern = regexPattern
+
                 evaluateSearchQueue.async(group: evaluateResultGroup) {
                     evaluateResultGroup.enter()
-                    Task {
-                        var newResult = SearchResultModel(file: CEWorkspaceFile(url: file.url), score: file.score)
-                        await self.evaluateFile(query: regexPattern, searchResult: &newResult)
-
-                        // Check if the new result has any line matches.
-                        if !newResult.lineMatches.isEmpty {
-                            // The function needs to be called because,
-                            // we are trying to modify the array from within a concurrent context.
-                            self.appendNewResultsToTempResults(newResult: newResult)
+                    Task { [weak self] in
+                        guard let self else {
+                            evaluateResultGroup.leave()
+                            return
+                        }
+
+                        let result = await self.evaluateSearchResult(
+                            fileURL: fileURL,
+                            fileScore: fileScore,
+                            regexPattern: capturedRegexPattern
+                        )
+
+                        if let result = result {
+                            await MainActor.run {
+                                self.tempSearchResults.append(result)
```

Comment:
Is there a reason this isn't using `appendNewResultsToTempResults`?

Reply:
Using it now